packages/common/src/ERC4337Utils.ts

```diff
@@ -45,8 +45,10 @@ export type NotPromise<T> = {
  * @param forSignature "true" if the hash is needed to calculate the getUserOpHash()
  *  "false" to pack entire UserOp, for calculating the calldata cost of putting it on-chain.
  */
-export function packUserOp(op: NotPromise<UserOperationStruct>, forSignature = true): string {
+export function packUserOp(op: Partial<UserOperation>, forSignature = true): string {
   if (forSignature) {
+    if (!op.initCode || !op.callData || !op.paymasterAndData)
+    op.initCode = op.callData = op.paymasterAndData = '0x'
```

Comment:
Why this check??

Even if one of the  initCode, callData and paymasterAndData is undefined. Why we assigning 0x to all the fields?